demos/text_detection_demo/cpp_gapi/src/text_recognition.cpp

```diff
@@ -0,0 +1,208 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <algorithm>
+#include <cmath>
+#include <numeric>
+#include <fstream>
+
+#include "text_recognition.hpp"
+
+bool fileExists(const std::string& filename) {
+    std::ifstream f(filename.c_str());
+    return f.good();
+}
+bool tryReadVocabFile(const std::string& filename, std::string& alphabet) {
+    if (fileExists(filename)) {
+        alphabet = "";
+        std::ifstream inputFile(filename);
+        if (!inputFile.is_open()) {
+            throw std::runtime_error("Can't open the vocab file: " + filename);
+        }
+        std::string vocabLine;
+        while (std::getline(inputFile, vocabLine)) {
+            if (vocabLine.length() != 1) {
+                throw std::invalid_argument("Lines in the vocab file must contain 1 character");
+            }
+            alphabet += vocabLine;
+        }
+        if (alphabet.empty()) {
+            throw std::logic_error("File is empty: " + filename);
+        }
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void softmaxAndChoose(const std::vector<float>::const_iterator& begin,
+                      const std::vector<float>::const_iterator& end,
+                      int *argmax, float *prob) {
+    auto maxElem = std::max_element(begin, end);
+    *argmax = static_cast<int>(std::distance(begin, maxElem));
+    float maxVal = *maxElem;
+    double sum = 0;
+    for (auto i = begin; i != end; i++) {
+        sum += std::exp((*i) - maxVal);
+    }
+    if (std::fabs(sum) < std::numeric_limits<double>::epsilon()) {
+        throw std::logic_error("sum can't be equal to zero");
+    }
+    *prob = 1.0f / static_cast<float>(sum);
+}
+
+std::vector<float> softmax(const std::vector<float>::const_iterator& begin,
+                           const std::vector<float>::const_iterator& end) {
+    std::vector<float> prob(end - begin, 0.f);
+    std::transform(begin, end, prob.begin(), [](float x) { return std::exp(x); });
+    float sum = std::accumulate(prob.begin(), prob.end(), 0.0f);
+    for (int i = 0; i < static_cast<int>(prob.size()); i++)
```

Comment:
for (auto& el : prob) {
    el /= sum;
}

Reply:
I guess

I didn't look to these functions deep as they are almost untouched-ly copied from `cpp` demo.